experiment/create-build-project.sh

```diff
@@ -0,0 +1,213 @@
+#!/usr/bin/env bash
+
+# Copyright 2019 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script will:
+#   Create a new GCP project
+#   Link a billing account
+#   Enable GKE
+#   Create a default build cluster
+
+# e.g.
+#  create-build-cluster.sh "Test Prow Build" --interactive
+
+# TODO(clarketm): upgrade this to a Python script.
+
+set -euo pipefail
+
+######################################################################
+
+SCRIPT_NAME=$(basename "$0")
+
+# Save current kube context
+CURRENT_CONTEXT="$(kubectl config current-context)"
+trap 'restore_kube_context' EXIT
```

Comment:
we should probably define `restore_kube_context` before trapping on it, or at least closer to it.

Reply:
Moved the trap after the function definition and put it specifically where it is needed.